mantidproject/mantid
# ALC Axis Labels And External Plotting

**Description of work.**
I have moved the External Plotted used in Indirect to a common place so it can be used with the ALC as well. I have made some changes to the external plotted to allow the use of kwargs. I have chosen not to add this for all the functions as this can be added as and when needed in the future. I have also added a text label to the PlotAsymmetryByLogValue workspace for the X axis. This means when it is plotted both in the ALC and from the external plot the x axis continues to have a label.

**To test:** 
(Each external plot should be identical to the plots from the ALC. Also whilst doing the steps check the x axis has a label where needed and the external plots have axis labels)
Open the ALC
Load MUSR62260-8
Click External Plot
Go to the Baseline Modelling tab
Add a section to fit by
Add a fit function such as Linear Background
Click fit
Click external Plot (here the Baseline Model plot is plotted as the Correct Data can be externally plotted from the next tab)
Go to the Peak Fitting Tab
Add a fit function, Polynomial for example
Click fit
Click External Plot
Go to the indirect tab and check the external plot features still work there as well.

Fixes #31479 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
